app/controllers/orders_controller.rb

```diff
@@ -14,12 +14,21 @@ def create
         new_quantity = (item.inventory - quantity)
         item_to_pull = Item.find(item.id)
         item_to_pull.update(inventory: new_quantity)
-        order.item_orders.create({
-          item: item,
-          quantity: quantity,
-          price: item.price,
-          merchant_id: item_to_pull.merchant_id
-          })
+        if cart.has_discount(item)
+          order.item_orders.create({
+            item: item,
+            quantity: quantity,
+            price: item.discount_price,
```

Comment:
Sets as discounted price, but is referenced through item_order.price This is advantageous because I didn't have to change anything anywhere else.